pkg/controllers/deprovisioning/controller.go

```diff
@@ -297,26 +301,38 @@ func (c *Controller) waitForDeletion(ctx context.Context, nodeClaim *v1beta1.Nod
 	}
 }
 
-func (c *Controller) setNodesUnschedulable(ctx context.Context, isUnschedulable bool, candidates ...*Candidate) error {
+// requireDisruptingNoScheduleTaint will either add/remove the karpenter.sh/disrupting taint from the candidates.
+// This is used to enforce no taints at the beginning of deprovisioning, and
+// to add/remove taints while executing a deprovisioning action.
+func (c *Controller) requireDisruptingNoScheduleTaint(ctx context.Context, addTaint bool, nodes ...*state.StateNode) error {
```

Comment:
I'm personally not a big fan of a single function that conveys vastly different behavior dictated by a flag. In this case we are using an `addTaint` flag to determine whether we remove or add taints from the nodes. Does it make more sense to just break this into two separate functions to make this more explicit?